bcl2fastq/index_summary.py

```diff
@@ -201,50 +127,43 @@ def update_known_index_bar(run_json):
                 'color': 'rgb(255,255, 255)'
             }}
         })
-
     return {
-        'data': data,
+        'data': data_known,
         'layout': {
             'barmode': 'stack',
             'title': 'Sample Indices',
             'xaxis': {'title': 'Library', 'automargin': True},
             'yaxis': {'title': 'Clusters'},
-        }
-    }
+        }}
 
 
-@app.callback(
-    dep.Output('unknown_index_bar', 'figure'),
-    [dep.Input('pruned_unknown_hidden', 'children')]
-)
-def update_unknown_index_bar(run_json):
-    """ When input (sample_run_hidden) is changed, function is rerun to get updated bar graph of known and unknown indices
-               Parameters:
-                   run_json(str): json format of run data
-
-               Returns:
-                   data and layout values for stacked bar graph for unknown indices
-                   updates unknown_index_bar bar graph
-           """
-    # pruned(str): holds first 30 unknown indices
-    pruned = pandas.read_json(run_json, orient='split')
+def update_unknown_index_bar(run_alias):
+    """ Function to update unknown index bar  according to user selected run
+            Parameters:
+                 run_alias: name of run
+            Returns:
+                data and layout values for stacked bar graph for unknown indices
+                updates unknown_index_bar bar graph
+              """
+    pruned = gsiqcetl.bcl2fastq.utility.prune_unknown_index_from_run(
+        run_alias, index, unknown
+    )
```

Comment:
Creating the `pruned` DataFrame is done twice (on line 189). Move the calculation into `update_layout` and pass the DataFrame.